react/index.native.js

```diff
@@ -1,3 +1,27 @@
+// The type field of react-native application loader's React Element is created
+// as number and not Symbol, because it's not been defined by the polyfill yet.
+// We import the application renderer, before Symbol is defined, in order to use
+// number types as well. Otherwise this will result in the invariant exception,
+// because fiber thingy will not recognise root react-native component as React
+// Element, but as an Object.
+//
+// See node_modules/react-native/Libraries/polyfills/babelHelpers.js
+// :babelHelpers.createRawReactElement - that's where first react-native element
+// is created (super early - it's the app loader).
+//
+// See node_modules/react-native/Libraries/Renderer/ReactNativeFiber-dev.js
+// and look for REACT_ELEMENT_TYPE definition - it's defined later when Symbol
+// has been defined and type will not match.
+//
+// As an alternative solution we could stop using/polyfilling Symbols and
+// replace with classpath string constants or some kind of a wrapper around
+// that.
+import 'react-native/Libraries/ReactNative/renderApplication';
+
+// FIXME: remove once react-native-webrtc and react-native-prompt imports
+// PropTypes from the 'prop-types' lib instead of 'react'.
+import './features/base/react/prop-types-polyfill.js';
+
 import 'es6-symbol/implement';
```

Comment:
@paweldomas, could you please clarify the following:

1. Today you told me that react-native on iOS now has `Symbol`. Which means that `import 'es6-symbol/implement';` is not necessary on iOS?

2. If the answer to 1. is positive, does it mean that `import 'react-native/Libraries/ReactNative/renderApplication';` is only necessary on Android then?

3. If the answer to 2. is positive, then why don't we have `import 'react-native/Libraries/ReactNative/renderApplication';` and `import 'es6-symbol/implement';` in index.android.js where there's practically no source code at the moment?

Reply:
@lyubomir That seems correct. iOS >= 9 has Symbol, and that's what we currently support. So it's a problem on Android only AFAICT. I'll be testing this today.